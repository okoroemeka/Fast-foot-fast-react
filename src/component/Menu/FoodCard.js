import React from "react";
const FoodCard = props => {
  return (
    <div className="col-3 food-card">
      <div className="food-card-child">
        <img
          src={props.food_image}
          alt={props.food}
          className="menu-food-image"
        />
        <div className="col-12 food-details">
          <ul>
            <li>
              <span className="facts">Name:</span>
              <span className="food">{props.food}</span>
            </li>
            <li>
              <span className="facts">Price:</span>
              <span className="Price"> &#8358; {props.price}</span>
            </li>
          </ul>
          <div className="row">
            <div className="col-6 order-button">
              <button className="order-now" type="submit" onClick={()=>props.onClickHandleOrderNow(props.meal)}>
                Order now
              </button>
            </div>
            <div className="col-6 add-to-cart-button">
              <button className="add-to-cart" onClick={()=>props.onClickAddToCart(props.meal)} type="submit">
                Add to cart
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default FoodCard;
