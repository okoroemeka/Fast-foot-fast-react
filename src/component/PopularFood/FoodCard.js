import React from "react";

const foodCard = props => (
  <div className="card">
    <div className="col-12">
      <img src={props.mealImage} className="food-image" alt={props.foodName} />

      <div className="row food-descrition-container">
        <div className="col-7 food-descrition">
          <ul className="food-name-ul">
            <li>
              <span className="food-name">{props.foodName}</span>
            </li>
          </ul>
        </div>
        <div className="col-5 food-descrition">
          <ul>
            <li>
              <span className="Price"> &#8358;{props.price}</span>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
);

export default foodCard;
