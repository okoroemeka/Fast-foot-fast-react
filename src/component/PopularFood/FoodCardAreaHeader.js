import React from 'react';
const FoodCardAreaHeader = () => {
  return ( 
    <div className='col-12 section-caption'>
      <p>Popular Food</p>
    </div>
   );
}
 
export default FoodCardAreaHeader;