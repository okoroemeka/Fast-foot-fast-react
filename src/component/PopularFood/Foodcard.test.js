import React from 'react';
import 'react-testing-library/cleanup-after-each';
import 'jest-dom/extend-expect';
import { render, cleanup } from 'react-testing-library';
import FoodCardAreaHeader from './FoodCardAreaHeader';
import FoodCard from './FoodCard';



afterEach(cleanup);

describe('<FoodCardAreaHeader />', () => {
  test('render correctly', () => {
    const  foodCardAreaHeader  = render(<FoodCardAreaHeader />);
    expect(foodCardAreaHeader).toMatchSnapshot();
  });
});

describe('<FoodCard />', () => {
  test('render correctly', () => {
    const  foodCard  = render(<FoodCard />);
    expect(foodCard).toMatchSnapshot();
  });
});