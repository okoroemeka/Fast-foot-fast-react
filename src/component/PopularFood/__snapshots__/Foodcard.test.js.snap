// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FoodCard /> render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="card"
      >
        <div
          class="col-12"
        >
          <img
            class="food-image"
          />
          <div
            class="row food-descrition-container"
          >
            <div
              class="col-7 food-descrition"
            >
              <ul
                class="food-name-ul"
              >
                <li>
                  <span
                    class="food-name"
                  />
                </li>
              </ul>
            </div>
            <div
              class="col-5 food-descrition"
            >
              <ul>
                <li>
                  <span
                    class="Price"
                  >
                     ₦
                  </span>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="card"
    >
      <div
        class="col-12"
      >
        <img
          class="food-image"
        />
        <div
          class="row food-descrition-container"
        >
          <div
            class="col-7 food-descrition"
          >
            <ul
              class="food-name-ul"
            >
              <li>
                <span
                  class="food-name"
                />
              </li>
            </ul>
          </div>
          <div
            class="col-5 food-descrition"
          >
            <ul>
              <li>
                <span
                  class="Price"
                >
                   ₦
                </span>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<FoodCardAreaHeader /> render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="col-12 section-caption"
      >
        <p>
          Popular Food
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="col-12 section-caption"
    >
      <p>
        Popular Food
      </p>
    </div>
  </div>,
  "debug": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
